src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/TriviaEngine/AbstractTriviaFormatter.cs

```diff
@@ -279,27 +289,41 @@ private LineColumn FormatTrivia<T>(Formatter<T> formatter, WhitespaceAppender<T>
             var implicitLineBreak = false;
 
             var list = new TriviaList(this.Token1.TrailingTrivia, this.Token2.LeadingTrivia);
+            var i = -1;
+            var previousLineColumn = new LineColumn();
```

Comment:
can you doc what this is for?

Reply:
Done

Reply:
💭 It feels like this local variable is hard-coding logic that is designed to be provided by implementations of `GetLineColumnRuleBetween` and/or `GetAdjustSpacesOperation` (which are called by `FormatFirstTriviaAndWhitespaceAfter` below). If `GetLineColumnRuleBetween`/`GetAdjustSpacesOperation` return the correct value, then `FormatFirstTriviaAndWhitespaceAfter` will also return the correct value and it looks like _all_ of the changes in this file become unnecessary (some of the logic would move into the space adjustment rule, but would no longer be applied in such a general location).